### PR TITLE
combined investigation search

### PR DESCRIPTION
**Pull request by kennyevil**
## Description

- Ensures export and search are finding same investigations.
- Adds new notification search
- Removes 10000 case limit for case export

https://regulatorydelivery.atlassian.net/jira/software/c/projects/PSD/boards/57?selectedIssue=PSD-2058
https://regulatorydelivery.atlassian.net/jira/software/c/projects/PSD/boards/57?selectedIssue=PSD-2082
https://regulatorydelivery.atlassian.net/jira/software/c/projects/PSD/boards/57?selectedIssue=PSD-2084
## Screen-shots or screen-capture of UI changes

Screenshot 2023-11-15 at 08-55-52 All Cases – Search - Product Safety Database - GOV UK

## Review apps

https://psd-pr-2704.london.cloudapps.digital/
https://psd-pr-2704-support.london.cloudapps.digital/

## Checklist:
- [x] Have you documented your changes in the pull request description?
- [ ] Has acceptance criteria been tested by a peer?

### General testing (author)
- [x] Test without JavaScript
- [x] Test on small screen

### Accessibility testing (author)
- [x] Reviewed by Designer (if required)
- [x] Works keyboard only
- [x] Tested with one screen reader
- [x] Zoom page to 400% - content still visible
- [x] Disable CSS - does content make sense and appear in a logical order?

### CODE REVIEW

File: app/controllers/searches_controller.rb [hunk 1]
```diff
@@ -4,10 +4,20 @@ class SearchesController < ApplicationController
...
     if @search.q.blank?
-      redirect_to investigations_path(query_params.except(:page_name))
+      if current_user.can_access_new_search?
+        redirect_to notifications_path(query_params.except(:page_name))
+      else
+        redirect_to investigations_path(query_params.except(:page_name))
+      end
     else
-      @answer = opensearch_for_investigations(20)
+      @answer = notfications_search
```
Comment (rubenarakelyan) [inline app/controllers/searches_controller.rb:13]: `notifications_search` :)

Comment (kennyevil) [reply app/controllers/searches_controller.rb:13]: I'll petition the dictionary to make the appropriate changes.

File: config/routes.rb [hunk 2]
```diff
@@ -111,6 +111,14 @@
 
     get "cases/new", to: "ts_investigations#new"
 
+    resource :notifications, only: %i[index] do
+      resource :search, only: :show
+
+      get "all-cases", to: "notifications#index"
+    end
+
+    resources :notifications, only: %i[index]
```
Comment (rubenarakelyan) [inline config/routes.rb:120]: Do we need both of these?

**Review by rubenarakelyan — APPROVED** [review]

**Review by AndrewCowle — APPROVED** [review]

File: config/routes.rb (hunk 2 above)
Comment (kennyevil) [reply config/routes.rb:118]: We're not using the "all-cases" path at the moment so I've removed it.

**Pull request merged**